hazelcast/src/main/java/com/hazelcast/internal/cluster/impl/ClusterServiceImpl.java

```diff
@@ -1008,14 +1004,21 @@ public void promoteLocalLiteMember() {
                 throw new IllegalStateException("Cannot promote to data member! Previous master was: " + master.getAddress()
                     + ", Current master is: " + getMasterAddress());
             }
-
-            localMember = new MemberImpl(member.getAddress(), member.getVersion(), true, member.getUuid(),
-                    member.getAttributes(), false, member.getMemberListJoinVersion(), node.hazelcastInstance);
         } finally {
             lock.unlock();
         }
     }
 
+    MemberImpl promoteAndGetLocalMember() {
+        MemberImpl member = getLocalMember();
+        assert member.isLiteMember() : "Local member is not lite member!";
+        assert lock.isHeldByCurrentThread() : "Called without holding cluster service lock!";
+
+        localMember = new MemberImpl(member.getAddress(), member.getVersion(), true, member.getUuid(),
```

Comment:
Since we are replacing the `localMember` here, we should also update the reference known to `node.loggingService`

Reply:
fixed